### PR TITLE
Add client functions for Operator Policy endpoints

### PR DESCRIPTION
**Pull request by awills96**
Also updated test files to add operator policy Api tests.

### CODE REVIEW

File: src/test/groovy/com/rabbitmq/http/client/ClientSpec.groovy [hunk 1]
```diff
@@ -1156,6 +1156,33 @@ class ClientSpec extends Specification {
 
   }
 
+  def "PUT /api/operator-policies/{vhost}/{name}"() {
+    given: "vhost / and definition"
+    def v = "/"
+    def d = new HashMap<String, Object>()
+    d.put("ha-mode", "all")
```
Comment (michaelklishin) [inline src/test/groovy/com/rabbitmq/http/client/ClientSpec.groovy:1163]: Please replace this with something like queue length limit. [Classic mirrored queues are going away completely in the next 3-9 months](https://rabbitmq.com/ha.html).

File: src/test/groovy/com/rabbitmq/http/client/ClientSpec.groovy [hunk 2]
```diff
@@ -2205,6 +2232,63 @@ class ClientSpec extends Specification {
 
   }
 
+  def "GET /api/operator-policies"() {
+    given: "at least one operator policy was declared"
+    def v = "/"
+    def s = "hop.test"
+    def d = new HashMap<String, Object>()
+    def p = ".*"
+    d.put("ha-mode", "all")
```
Comment (michaelklishin) [inline src/test/groovy/com/rabbitmq/http/client/ClientSpec.groovy:2241]: Same as above

File: src/test/groovy/com/rabbitmq/http/client/ClientSpec.groovy [hunk 3]
```diff
@@ -2205,6 +2232,63 @@ class ClientSpec extends Specification {
...
+  }
+
+
+  def "GET /api/operator-policies/{vhost} when vhost exists"() {
+    given: "at least one operator  was declared in vhost /"
+    def v = "/"
+    def s = "hop.test"
+    def d = new HashMap<String, Object>()
+    def p = ".*"
+    d.put("ha-mode", "all")
```
Comment (michaelklishin) [inline src/test/groovy/com/rabbitmq/http/client/ClientSpec.groovy:2263]: Same as above

File: src/test/groovy/com/rabbitmq/http/client/ReactorNettyClientSpec.groovy [hunk 4]
```diff
@@ -1109,6 +1109,59 @@ class ReactorNettyClientSpec extends Specification {
         exception.status() == 404
     }
 
+    def "GET /api/operator-policies"() {
+        given: "at least one operator policy was declared"
+        def v = "/"
+        def s = "hop.test"
+        def d = new HashMap<String, Object>()
+        def p = ".*"
+        d.put("ha-mode", "all")
```
Comment (michaelklishin) [inline src/test/groovy/com/rabbitmq/http/client/ReactorNettyClientSpec.groovy:1118]: Same as above

File: src/test/groovy/com/rabbitmq/http/client/ReactorNettyClientSpec.groovy [hunk 5]
```diff
@@ -1109,6 +1109,59 @@ class ReactorNettyClientSpec extends Specification {
...
+        client.deleteOperatorPolicy(v, s).block()
+    }
+
+    def "GET /api/operator-policies/{vhost} when vhost exists"() {
+        given: "at least one operator policy was declared in vhost /"
+        def v = "/"
+        def s = "hop.test"
+        def d = new HashMap<String, Object>()
+        def p = ".*"
+        d.put("ha-mode", "all")
```
Comment (michaelklishin) [inline src/test/groovy/com/rabbitmq/http/client/ReactorNettyClientSpec.groovy:1138]: Same as above

**Review by michaelklishin — CHANGES_REQUESTED** [review]

**Comment by michaelklishin** [comment]
The rest of the PR looks good to me. Thank you!

**Comment by michaelklishin** [comment]
Thank you!

**Comment by acogoluegnes** [comment]
@awills96 Thanks for for this contribution. Would you mind [signing the CLA](https://github.com/rabbitmq/hop/pull/396#issuecomment-1716085822)? Thanks!

**Comment by awills96** [comment]
Thanks, will do soon! I'm just waiting on some details and managerial approvals since I did do this on corporate behalf.

**Pull request merged**